encoding/string/encoding.go

```diff
@@ -0,0 +1,46 @@
+package string
+
+import "io"
+
+// Encoder encodes string values.
+type Encoder interface {
```

Comment:
Can you create an issue for templatizing these interfaces if they are exactly the same aside from the type difference?

Reply:
Yeah nw, I'm used to creating issues for things that do not require immediate attention but may be worth revisiting at some point.

Reply:
https://github.com/xichen2020/eventdb/issues/14